src/schema/v2/me/myCollectionUpdateArtworkMutation.ts

```diff
@@ -104,22 +130,36 @@ export const myCollectionUpdateArtworkMutation = mutationWithClientMutationId<
       })
 
       if (!response.edition_sets?.length) {
-        if (editionSize || editionNumber) {
-          // create new edition set when none existed previously
-          await createArtworkEditionSetLoader(artworkId, {
-            edition_size: editionSize,
-            available_editions: editionNumber ? [editionNumber] : null,
-          })
+        const providedEditionField = editionNumber || editionSize
+        if (isEdition === true || providedEditionField) {
+          // create edition set for artwork
+          let payload = {}
+          if (editionSize) {
+            payload["edition_size"] = editionSize
+          }
+
+          if (editionNumber) {
+            payload["available_editions"] = [editionNumber]
+          }
+          await createArtworkEditionSetLoader(artworkId, payload)
         }
       } else {
         const editionSetId = response.edition_sets[0].id
-        await updateArtworkEditionSetLoader(
-          { artworkId, editionSetId },
-          {
-            edition_size: editionSize,
-            available_editions: editionNumber ? [editionNumber] : null,
+
+        if (isEdition === false) {
+          await deleteArtworkEditionSetLoader({ artworkId, editionSetId })
+        } else {
+          const payload = {
+            edition_size: editionSize ? editionSize : null,
+            // TODO: Is there a better way to clear out edition number?
+            available_editions: editionNumber ? [editionNumber] : [""],
```

Comment:
This is a bit weird, I can't find a way of clearing out `avaliable_editions` which populates `edition_number` cleanly. Passing null fails gravity validation, passing empty array does not clear out edition_number. Passing an array with empty string is close to what we are after making 'edition_number' an empty string but ideally I could pass something here to make the field null.

Reply:
I agree this is a bit weird. I'm happy to ship this as is and follow-up with a Gravity PR to allow `avaliable_editions` to be `null`. Might make a good KS topic!